cpp/src/utilities/wrapper_types.hpp

```diff
@@ -305,4 +348,45 @@ using date64 = detail::wrapper<gdf_date64, GDF_DATE64>;
 
 } // namespace cudf
 
+namespace std
+{
+
+/**---------------------------------------------------------------------------*
+ * @brief Specialization of std::numeric_limits for wrapper types
+ *---------------------------------------------------------------------------**/
+template <typename T, gdf_dtype type_id>
+struct numeric_limits< cudf::detail::wrapper<T, type_id> > {
+  
+  using wrapper_t = cudf::detail::wrapper<T, type_id>;
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns the maximum finite value representable by the numeric type T
+   *---------------------------------------------------------------------------**/
+  static constexpr wrapper_t max() noexcept {
+    return wrapper_t{ std::numeric_limits<T>::max() };
```

Comment:
So you are still basically accessing the underlying type of the wrapper. What if we want to provide different semantics for `max()` and `lowest()` for the wrapper types than for the underlying type?

Reply:
Then you'd specialize this template to handle each different wrapper type. Something on the lines of:
```c++
template <>
struct numeric_limits< cudf::date32 > {
  using wrapper_t = cudf::date32;
  static constexpr wrapper_t max() noexcept { ... }
  static constexpr wrapper_t lowest() noexcept { ... }
  static constexpr wrapper_t min() noexcept { ... }
};
```
That wasn't the case for the wrapper types that currently exist though. I checked for dates and their mix and max are indeed the min and max of the integer types of their underlying data. You might think this wouldn't be the case for category types but in a discussion yesterday with @kkraus14 and @felipeblazing , I found out it *was* the case. (They said that comparison operators (>,<,>=,<=,==) should work, which I took to assume that min and max are also defined in the same way)